main.cc

```diff
@@ -1801,22 +1834,6 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
 
             notify_set.notify_all(configurable::system_state::started).get();
 
-            scheduling_group_key_config cql_sg_stats_cfg = make_scheduling_group_key_config<cql_transport::cql_sg_stats>(maintenance_port_enabled::no);
-            scheduling_group_key_config maintenance_cql_sg_stats_cfg = make_scheduling_group_key_config<cql_transport::cql_sg_stats>(maintenance_port_enabled::yes);
-            cql_transport::controller cql_server_ctl(auth_service, mm_notifier, gossiper, qp, service_memory_limiter, sl_controller, lifecycle_notifier, *cfg, scheduling_group_key_create(cql_sg_stats_cfg).get0(), maintenance_port_enabled::no);
-            cql_transport::controller cql_maintenance_server_ctl(maintenance_auth_service, mm_notifier, gossiper, qp, service_memory_limiter, sl_controller, lifecycle_notifier, *cfg, scheduling_group_key_create(maintenance_cql_sg_stats_cfg).get0(), maintenance_port_enabled::yes);
-
-            auto start_cql = [&dbcfg] (cql_transport::controller& controller, std::any& stop_cql) {
-                supervisor::notify("starting native transport");
-                with_scheduling_group(dbcfg.statement_scheduling_group, [&controller] {
-                    return controller.start_server();
-                }).get();
-
-                // FIXME -- this should be done via client hooks instead
-                stop_cql = defer_verbose_shutdown("native transport", [&controller] {
-                    controller.stop_server().get();
-                });
-            };
             ss.local().register_protocol_server(cql_server_ctl);
```

Comment:
I don't understand this patch, it says it adds maintenance mode, but it moves around code that already handles maintenance mode.

Reply:
Discussion resolved in maintenance socket PR.